### PR TITLE
Fix limiting unlimited vector search

### LINKED ISSUE #1905: pagination limit not adhered to when using nearVector/Object/Text search
With the addition of unlimited vector search, Weaviate is able to exhaustively search all vectors within a specified certainty if not limit is provided to the query, or with a limit of `-1`.

However, it was determined by a Weaviate user that when running a query with a certainty and a (positive) limit, the provided limit was not adhered to and Weaviate returned all results beyond the specified limit.

### PR DESCRIPTION
**Pull request by parkerduckworth**
Fixes #1905

### CODE REVIEW

**Review by aliszka — APPROVED** [review]

File: adapters/handlers/graphql/local/get/class_builder_fields.go [hunk 1]
```diff
@@ -389,7 +389,7 @@ func setLimitBasedOnVectorSearchParams(params *traverser.GetParams) {
 			params.Pagination = &filters.Pagination{
 				Limit: filters.LimitFlagSearchByDist,
 			}
-		} else {
+		} else if params.Pagination.Limit < 0 {
```
Comment (parkerduckworth) [inline adapters/handlers/graphql/local/get/class_builder_fields.go:392]: This is the one-line fix for the limiting issue

File: test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go [hunk 2]
```diff
@@ -0,0 +1,338 @@
...
+	"github.com/semi-technologies/weaviate/test/acceptance/helper"
+	"github.com/stretchr/testify/require"
+)
+
+func gettingObjectsWithNearFields(t *testing.T) {
+	defaultLimit := 100
+
+	// nearVector
+
+	t.Run("nearVector: with implicit unlimited search - no limit provided", func(t *testing.T) {
```
Comment (parkerduckworth) [inline test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go:16]: @etiennedi implicit unlimited search (no limit provided)

File: test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+			}
+		}
+		`
+		result := AssertGraphQL(t, helper.RootAuth, query)
+		notes := result.Get("Get", "RansomNote").AsSlice()
+		require.NotEmpty(t, notes)
+		require.Greater(t, len(notes), defaultLimit)
+	})
+
+	t.Run("nearVector: with implicit unlimited search - negative limit provided", func(t *testing.T) {
```
Comment (parkerduckworth) [inline test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go:40]: @etiennedi explicit unlimited search (limit passed as -1)

File: test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go [hunk 4]
```diff
@@ -0,0 +1,338 @@
...
+			}
+		}
+		`
+		result := AssertGraphQL(t, helper.RootAuth, query)
+		notes := result.Get("Get", "RansomNote").AsSlice()
+		require.NotEmpty(t, notes)
+		require.Greater(t, len(notes), defaultLimit)
+	})
+
+	t.Run("nearVector: with limited search - limit provided", func(t *testing.T) {
```
Comment (parkerduckworth) [inline test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go:65]: @etiennedi limited search (non-negative limit passed)

File: test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go [hunk 5]
```diff
@@ -0,0 +1,338 @@
...
+			}
+		}
+		`
+		result := AssertGraphQL(t, helper.RootAuth, query)
+		notes := result.Get("Get", "RansomNote").AsSlice()
+		require.NotEmpty(t, notes)
+		require.Equal(t, 10, len(notes))
+	})
+
+	t.Run("nearVector: results limited by certainty", func(t *testing.T) {
```
Comment (parkerduckworth) [inline test/acceptance/graphql_resolvers/local_get_with_unlimited_vector_search.go:90]: @etiennedi unlimited search , limited by certainty (no limit passed, but certainty set very high to limit results to less than default limit)

**Review by parkerduckworth — COMMENTED** [review]
@etiennedi each of the three `near*` fields have these 4 test cases

**Review by antas-marcin — APPROVED** [review]

**Pull request merged**